src/fsharp/service/service.fs

```diff
@@ -1855,22 +1855,23 @@ type FSharpCheckProjectResults(projectFileName:string, tcConfigOption, keepAssem
     member info.GetUsesOfSymbol(symbol:FSharpSymbol) = 
         let (tcGlobals, _tcImports, _thisCcu, _ccuSig, tcSymbolUses, _topAttribs, _tcAssemblyData, _ilAssemRef, _ad, _tcAssemblyExpr, _dependencyFiles) = getDetails()
 
-        [| for r in tcSymbolUses do yield! r.GetUsesOfSymbol(symbol.Item) |] 
-        |> Seq.distinctBy (fun (itemOcc,_denv,m) -> itemOcc, m) 
-        |> Seq.filter (fun (itemOcc,_,_) -> itemOcc <> ItemOccurence.RelatedText) 
-        |> Seq.map (fun (itemOcc,denv,m) -> FSharpSymbolUse(tcGlobals, denv, symbol, itemOcc, m)) 
+        tcSymbolUses
+        |> Seq.collect (fun r -> r.GetUsesOfSymbol symbol.Item)
+        |> Seq.distinctBy (fun symbolUse -> symbolUse.ItemOccurence, symbolUse.Range) 
+        |> Seq.filter (fun symbolUse -> symbolUse.ItemOccurence <> ItemOccurence.RelatedText) 
+        |> Seq.map (fun symbolUse -> FSharpSymbolUse(tcGlobals, symbolUse.DisplayEnv, symbol, symbolUse.ItemOccurence, symbolUse.Range)) 
```

Comment:
A `choose` would slot in here, Im not sure how many iterations occurs in the global section there always a few hundred to iterate over.

Reply:
yeah wsn't sure if it's worth it here